chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -55,4 +55,52 @@ ovs-vsctl set-fail-mode <%= bridge %> standalone
 echo "Reloading wicked configuration for all interfaces"
 wicked ifreload all
 
+<% if @compute_node %>
+
+echo "Upgrading services needed to act as nova-compute node"
+
+zypper --non-interactive up openstack-nova-compute
+# This is ugly, but fixing neutron spec to upgrade to correct version of pecan is even uglier:
+zypper --non-interactive up python-pecan
+zypper --non-interactive install crudini
+
+echo "Upgrading configuration for Newton based packages"
+
+crudini --set /etc/nova/nova.conf.d/200-crowbar-upgrade.conf DEFAULT use_neutron true
+
+zypper --non-interactive up <%= @neutron_agent %>
+<% case @neutron_agent %>
+<% when "openstack-neutron-openvswitch-agent" %>
+crudini --set /etc/neutron/neutron-openvswitch-agent.conf.d/200-crowbar-upgrade.conf ovs ovsdb_interface vsctl
+crudini --set /etc/neutron/neutron-openvswitch-agent.conf.d/200-crowbar-upgrade.conf ovs of_interface ovs-ofctl
+<% when "openstack-neutron-linuxbridge-agent" %>
+# FIXME: configuration changes?
```

Comment:
I believe in the processed file on the destination node this looks really weird. I guess it will be something like:

```
crudini --set /etc/neutron/neutron-openvswitch-agent.conf.d/200-crowbar-upgrade.conf ovs ovsdb_interface vsctl
crudini --set /etc/neutron/neutron-openvswitch-agent.conf.d/200-crowbar-upgrade.conf ovs of_interface ovs-ofctl
# FIXME: configuration changes?
systemctl start neutron-openvswitch-agent
```

As the comment is actually meant for the template code (mainly for the ```when``` statement), I guess it would make sense to not have it appear in the template's output. I'd suggest to used ```<%# comment %> here. Same of the other FIXMEs as well.